pwndbg/typeinfo.py

```diff
@@ -65,17 +65,20 @@ def update():
     module.int32  = lookup_types('int', 'i32', 'int32')
     module.int64  = lookup_types('long long', 'long', 'i64', 'int64')
 
-    module.ssize_t = module.long
-    module.size_t = module.ulong
-
     module.pvoid  = void.pointer()
     module.ppvoid = pvoid.pointer()
     module.pchar  = char.pointer()
 
     module.ptrsize = pvoid.sizeof
 
-    if pvoid.sizeof == 4: module.ptrdiff = uint32
-    if pvoid.sizeof == 8: module.ptrdiff = uint64
+    if pvoid.sizeof == 4: 
+        module.ptrdiff = module.uint32
+        module.size_t = module.uint32
+        module.ssize_t = module.int32
+    if pvoid.sizeof == 8: 
```

Comment:
```suggestion
    elif pvoid.sizeof == 8: 
```

Reply:
Can we also add an `else` with warning/error, just as a sanity check?

Reply:
Good call. We could also potentially support 16 bit binaries here, although I'm not sure if we want to do that, especially without testing more.